cdap-data-fabric/src/main/java/co/cask/cdap/data2/audit/KafkaAuditPublisher.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.audit;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.http.SecurityRequestContext;
+import co.cask.cdap.proto.audit.AuditMessage;
+import co.cask.cdap.proto.audit.AuditPayload;
+import co.cask.cdap.proto.audit.AuditType;
+import co.cask.cdap.proto.id.EntityId;
+import com.google.common.base.Charsets;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import org.apache.twill.kafka.client.Compression;
+import org.apache.twill.kafka.client.KafkaClient;
+import org.apache.twill.kafka.client.KafkaPublisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Publish audit messages on Kafka.
+ */
+public class KafkaAuditPublisher implements AuditPublisher {
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaAuditPublisher.class);
+  private static final Gson GSON = new Gson();
+
+  private final Supplier<KafkaPublisher> publisherSupplier;
+  private final String kafkaTopic;
+
+  @Inject
+  public KafkaAuditPublisher(final KafkaClient kafkaClient,
+                             @Named(Constants.Audit.KAFKA_TOPIC) String kafkaTopic) {
+    this.publisherSupplier = Suppliers.memoize(new Supplier<KafkaPublisher>() {
+      @Override
+      public KafkaPublisher get() {
+        return kafkaClient.getPublisher(KafkaPublisher.Ack.LEADER_RECEIVED, Compression.SNAPPY);
+      }
+    });
+    this.kafkaTopic = kafkaTopic;
+  }
+
+  @Override
+  public void publish(EntityId entityId, AuditType auditType, AuditPayload auditPayload) {
+    // TODO: what should user be when security is not enabled?
```

Comment:
Discussed this with @anew. We decided to go with empty string when remote user is not known when security is not enabled. I'll remove this TODO.